src/components/legend/style.js

```diff
@@ -104,6 +114,28 @@ module.exports = function style(s, gd) {
         var showGradientFill = false;
         var dMod, tMod;
 
+        var colorscale = getColorscale(trace);
+        var reversescale = getReversescale(trace);
```

Comment:
Can we reuse `extractOpts` instead 

https://github.com/plotly/plotly.js/blob/8951847f7edecd674cfebd2b670b550a97396745/src/components/colorscale/helpers.js#L65-L106

Something like:

```js
var cOpts = extractOpts(trace);
var colorscale = cOpts.colorscale;
var reversescale = cOpts.reverscale;
```

Reply:
... and we wouldn't need `getColorscale` nor `getReversescale`

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4356/commits/f988a643517d388af38ccd9586a4ec2065eb711a.

Reply:
Awesome thanks!